torch/_inductor/scheduler.py

```diff
@@ -360,6 +360,80 @@ def codegen_originating_info(self, buffer, only_once=True):
         buffer.writelines(out_lines)
         self.written = True
 
+    def get_estimated_runtime(self, rwbytes: int) -> float:
+        layout = None
+        if not self.node:
+            assert(self.snodes)
+            layout = self.snodes[0].node.get_layout()
+        else:
+            layout = self.node.get_layout()
+
+        if "cuda" != layout.device.type:
+            # default to no reordering based on runtime
+            return 0
+
+        from .ir import CollectiveKernel, ComputedBuffer
+
+        # TODO(xmfan): figure out how to get hardware specs, use A100 for now
+        gpu_memory_bandwidth = 1555 * 2**30  # 1555 GBps
+        gpu_flops = 312**12  # 312 TFLOPS
+
+
+        def handle_extern_kernel(snode: ExternKernelSchedulerNode):
+            from torch.utils.flop_counter import (
```

Comment:
Would be cleaner to do this using faketensors and flopcountermode instead of directly calling each operator separately.

Reply:
Addressed in [ba0fdc2](https://github.com/pytorch/pytorch/pull/106426/commits/ba0fdc201a50fed3bbf66acbf20a1e6e507cb9f6)